web/src/gifs.ts

```diff
@@ -99,7 +100,8 @@ const tenor_rating_map = {
     "pg-13": "low",
 };
 
-export function get_base_payload(gif_provider: "tenor" | "giphy"): TenorPayload | GiphyPayload {
+export function get_base_payload(): TenorPayload | GiphyPayload {
+    assert(gif_provider !== undefined);
```

Comment:
Architecturally, it's probably better for functions like this to accept the `gif_provider` as an attribute.

Would make refactoring simpler in a not super likely future where we sometimes offer both search engines. But more importantly, just reducing reliance on global state is generally good for readability.